### PR TITLE
Improve the snapshot tool command handling

### PR DESCRIPTION
**Pull request by galpeter**
Reworked the command handling of the snapshot tool to
allow adding new commands in a simpler way.

JerryScript-DCO-1.0-Signed-off-by: Peter Gal pgal.u-szeged@partner.samsung.com

### CODE REVIEW

File: jerry-main/main-unix-snapshot.c [hunk 1]
```diff
@@ -32,6 +33,28 @@
 #define JERRY_STANDALONE_EXIT_CODE_OK   (0)
 #define JERRY_STANDALONE_EXIT_CODE_FAIL (1)
 
+/**
+ * Function type for command handling call
+ */
+typedef int (*command_function_t) (cli_state_t *cli_state_p, int argc, char *prog_name_p);
+/**
```
Comment (akosthekiss) [inline jerry-main/main-unix-snapshot.c]: style: add an empty line separator between the two typedefs

File: jerry-main/main-unix-snapshot.c [hunk 2]
```diff
@@ -32,6 +33,28 @@
...
+ * Description for a command entries
+ */
+typedef struct
+{
+  /** Name of the command to be used */
+  const char *name;
+  /** Function to call for this command */
```
Comment (akosthekiss) [inline jerry-main/main-unix-snapshot.c]: style: `/**< doc */` "after member" comments would be nicer

**Comment by zherczeg** [comment]
Is there active work on this?

**Comment by galpeter** [comment]
Sorry, I'll get back to this (eventually). I'll close this down until I've got something more useful.

**Pull request closed without merging**